File: 3sem/4lab/Banks/models/BankConfigurations/DebitBankAccountConfiguration.cs
using Banks.Exceptions;

namespace Banks.Models.BankConfigurations;

public record DebitBankAccountConfiguration
{
    public DebitBankAccountConfiguration(decimal unverifiedLimit, decimal interestRate)
    {
        if (interestRate is < 0 or > 1)
            throw InvalidBankAccountOperationException.OnInvalidInterestRate();

        if (unverifiedLimit < 0)
            throw InvalidBankAccountOperationException.OnInvalidCreditLimit();

        UnverifiedLimit = unverifiedLimit;
        InterestRate = interestRate;
    }

    public decimal UnverifiedLimit { get; }
    public decimal InterestRate { get; }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Models/BankConfigurations/DebitBankAccountConfiguration.java
package banksRemake.domain.models.bankConfigurations;

/**
 * A set of parameters characterizing the debit account.
 * @param unverifiedLimit maximum transaction amount for an unverified user.
 * @param interestRate interest accrued every month.
 */
public record DebitBankAccountConfiguration(double unverifiedLimit,
                                            double interestRate) {
}
